File: robot/motor.py
import subprocess

def stop():
    """Stopper begge motorer."""
    subprocess.run(["pinctrl", "set", "17", "op", "dl"])
    subprocess.run(["pinctrl", "set", "27", "op", "dl"])
    subprocess.run(["pinctrl", "set", "22", "op", "dl"])
    subprocess.run(["pinctrl", "set", "23", "op", "dl"])

def forward():
    """Kører fremad."""
    subprocess.run(["pinctrl", "set", "17", "op", "dh"])
    subprocess.run(["pinctrl", "set", "27", "op", "dl"])
    subprocess.run(["pinctrl", "set", "22", "op", "dh"])
    subprocess.run(["pinctrl", "set", "23", "op", "dl"])

def backward():
    """Bakker."""
    subprocess.run(["pinctrl", "set", "17", "op", "dl"])
    subprocess.run(["pinctrl", "set", "27", "op", "dh"])
    subprocess.run(["pinctrl", "set", "22", "op", "dl"])
    subprocess.run(["pinctrl", "set", "23", "op", "dh"])

def left():
    """Drej til venstre (venstre motor bakker, højre motor kører frem)."""
    subprocess.run(["pinctrl", "set", "17", "op", "dl"])
    subprocess.run(["pinctrl", "set", "27", "op", "dh"])
    subprocess.run(["pinctrl", "set", "22", "op", "dh"])
    subprocess.run(["pinctrl", "set", "23", "op", "dl"])

def right():
    """Drej til højre (venstre motor kører frem, højre motor bakker)."""
    subprocess.run(["pinctrl", "set", "17", "op", "dh"])
    subprocess.run(["pinctrl", "set", "27", "op", "dl"])
    subprocess.run(["pinctrl", "set", "22", "op", "dl"])
    subprocess.run(["pinctrl", "set", "23", "op", "dh"])

File: robot/src/motor.cpp
#include "../include/motor.hpp"
#include <cstdlib>

void motor_stop() {
    // Stopper begge motorer
    std::system("pinctrl set 17 op dl");
    std::system("pinctrl set 27 op dl");
    std::system("pinctrl set 22 op dl");
    std::system("pinctrl set 23 op dl");
}

void motor_forward() {
    // Kører fremad
    std::system("pinctrl set 17 op dh");
    std::system("pinctrl set 27 op dl");
    std::system("pinctrl set 22 op dh");
    std::system("pinctrl set 23 op dl");
}

void motor_backward() {
    // Bakker
    std::system("pinctrl set 17 op dl");
    std::system("pinctrl set 27 op dh");
    std::system("pinctrl set 22 op dl");
    std::system("pinctrl set 23 op dh");
}

void motor_left() {
    // Drej til venstre
    std::system("pinctrl set 17 op dl");
    std::system("pinctrl set 27 op dh");
    std::system("pinctrl set 22 op dh");
    std::system("pinctrl set 23 op dl");
}

void motor_right() {
    // Drej til højre
    std::system("pinctrl set 17 op dh");
    std::system("pinctrl set 27 op dl");
    std::system("pinctrl set 22 op dl");
    std::system("pinctrl set 23 op dh");
}